From: Baolin Wang <baolin.wang@linux.alibaba.com>
Subject: mm: migrate: drop redundant anonymous statistics for file folios migration
Date: Thu, 5 Dec 2024 11:53:48 +0800

Commit 5d65c8d758f2 ("mm: count the number of anonymous THPs per size")
adds a new anonymous counter per THP size, however, when folio_mapping()
is not NULL during folio migration, it means this is not an anonymous
folio, so remove the redundant anonymous statistics in this case.

Link: https://lkml.kernel.org/r/d5e191ae02c2ab63c70cdf44877fe14fc69f5ae8.1733368611.git.baolin.wang@linux.alibaba.com
Signed-off-by: Baolin Wang <baolin.wang@linux.alibaba.com>
Cc: David Hildenbrand <david@redhat.com>
Cc: Zi Yan <ziy@nvidia.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/migrate.c |    2 --
 1 file changed, 2 deletions(-)

--- a/mm/migrate.c~mm-migrate-drop-redundant-anonymous-statistics-for-file-folios-migration
+++ a/mm/migrate.c
@@ -523,8 +523,6 @@ static int __folio_migrate_mapping(struc
 	 */
 	newfolio->index = folio->index;
 	newfolio->mapping = folio->mapping;
-	if (folio_test_anon(folio) && folio_test_large(folio))
-		mod_mthp_stat(folio_order(folio), MTHP_STAT_NR_ANON, 1);
 	folio_ref_add(newfolio, nr); /* add cache reference */
 	if (folio_test_swapbacked(folio)) {
 		__folio_set_swapbacked(newfolio);
_
